test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1

```diff
@@ -0,0 +1,580 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
@iSazonov, your last commit had 13 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Hidden
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 205
205:             $result | Where-Object { $_ -eq "subDir2\filehidden2.asd" } | Should -Not -BeNullOrEmpty
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Force
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 215
215:             $result | Where-Object { $_ -eq "subDir2\filehidden2.asd" } | Should -Not -BeNullOrEmpty
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -File -Hidden
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 247
247:             $result | Where-Object { $_ -eq "subDir2\filehidden2.asd" } | Should -Not -BeNullOrEmpty
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter -Recurse -Name.Get-ChildItem -Path -Filter "*.txt" -Recurse -Name
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 344
344:             $result | Where-Object { $_ -eq "subDir2\file2.txt" } | Should -Not -BeNullOrEmpty
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter -Recurse -Name.Get-ChildItem -Path -Filter "file*" -Recurse -Name
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.163.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 353
353:             $result | Where-Object { $_ -eq "subDir2\file2.txt" } | Should -Not -BeNullOrEmpty
```